SmartDeviceLink/SDLCarWindow.m

```diff
@@ -0,0 +1,216 @@
+
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(sdl_willPresentLockScreenViewController:) name:SDLLockScreenManagerWillPresentLockScreenViewController object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(sdl_willDismissLockScreenViewController:) name:SDLLockScreenManagerWillDismissLockScreenViewController object:nil];
+
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(sdl_didPresentLockScreenViewController:) name:SDLLockScreenManagerDidPresentLockScreenViewController object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(sdl_didDismissLockScreenViewController:) name:SDLLockScreenManagerDidDismissLockScreenViewController object:nil];
+
+    return self;
+}
+
+- (void)syncFrame {
+    if (!self.streamManager.isVideoConnected || self.streamManager.isVideoStreamingPaused) {
+        return;
+    }
+
+    if (self.isLockScreenPresenting || self.isLockScreenDismissing) {
+        SDLLogD(@"Paused CarWindow, lock screen moving");
+        return;
+    }
+
+    CGRect bounds = self.rootViewController.view.bounds;
+
+    UIGraphicsBeginImageContextWithOptions(bounds.size, YES, 1.0f);
+    [self.rootViewController.view drawViewHierarchyInRect:bounds afterScreenUpdates:self.drawsAfterScreenUpdates];
+    UIImage *screenshot = UIGraphicsGetImageFromCurrentImageContext();
+    UIGraphicsEndImageContext();
+
+    CGImageRef imageRef = screenshot.CGImage;
+    CVPixelBufferRef pixelBuffer = [self.class sdl_pixelBufferForImageRef:imageRef usingPool:self.streamManager.pixelBufferPool];
+    [self.streamManager sendVideoData:pixelBuffer];
+    CVPixelBufferRelease(pixelBuffer);
+}
+
+#pragma mark - SDLNavigationLockScreenManager Notifications
+- (void)sdl_willPresentLockScreenViewController:(NSNotification *)notification {
+    self.lockScreenPresenting = YES;
+}
+
+- (void)sdl_didPresentLockScreenViewController:(NSNotification *)notification {
+    self.lockScreenPresenting = NO;
+}
+
+- (void)sdl_willDismissLockScreenViewController:(NSNotification *)notification {
+    self.lockScreenBeingDismissed = YES;
+}
+
+- (void)sdl_didDismissLockScreenViewController:(NSNotification *)notification {
+    self.lockScreenBeingDismissed = NO;
+}
+
+#pragma mark - SDLNavigationLifecycleManager Notifications
+- (void)sdl_didReceiveVideoStreamStarted:(NSNotification *)notification {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        // If the video stream has started, we want to resize the streamingViewController to the size from the RegisterAppInterface
+        self.rootViewController.view.frame = CGRectMake(0, 0, self.streamManager.screenSize.width, self.streamManager.screenSize.height);
+        self.rootViewController.view.bounds = self.rootViewController.view.frame;
+
```

Comment:
The `rootViewController` is not being resized correctly after reconnecting to the head unit if the app on the device is in the background.

1. Connect phone to TDK
1. Open app on TDK
1. Disconnect phone from TDK by pulling the USB cord
1. Put app on phone in the background
1. Reconnect the phone to the TDK using the USB cord
1. The dialog box with the spinner should show up (waiting for video)
1. Bring the app on the phone to the foreground

The `rootViewController`'s frame is not being set to the dimensions of the TDK screen so the video stream does not fill the entire screen.

Reply:
This turned out to be a user error on my part and not an issue with the library. In the example above I was setting the `rootViewController` using an on-screen view controller. If the `rootViewController` was set before the on-screen view controller rendered its layout, resizing the frame failed.

Reply:
Can you describe your fix? Is there any documentation we can add?

Reply:
I added documentation around this to the configuration of `rootViewController` and to the guide content.